cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy

```diff
@@ -0,0 +1,86 @@
+package org.bf2.cos.connector.camel.it.azure.storage.blob
+
+import com.azure.core.http.policy.HttpLogDetailLevel
+import com.azure.core.http.policy.HttpLogOptions
+import com.azure.core.util.BinaryData
+import com.azure.storage.blob.BlobContainerClient
+import com.azure.storage.blob.BlobServiceClientBuilder
+import com.azure.storage.common.StorageSharedKeyCredential
+import groovy.util.logging.Slf4j
+import org.bf2.cos.connector.camel.it.support.KafkaConnectorSpec
+import spock.lang.IgnoreIf
+
+import java.time.Instant
+
+@IgnoreIf({
+    !hasEnv('AZURE_BLOB_ACCOUNT_NAME'   ) ||
+    !hasEnv('AZURE_BLOB_ACCESS_KEY'     ) ||
+    !hasEnv('AZURE_BLOB_CONTAINER_NAME' )
+})
+@Slf4j
+class AzureStorageBlobChangefeedConnectorIT extends KafkaConnectorSpec {
+
+    def "azure-storage-blob-changefeed source"() {
+        setup:
+            def topic = topic()
+            def group = UUID.randomUUID().toString()
+            def payload = topic
+
+            def containerName = System.getenv('AZURE_BLOB_CONTAINER_NAME')
+            def accountName = System.getenv('AZURE_BLOB_ACCOUNT_NAME')
+            def accountKey = System.getenv('AZURE_BLOB_ACCESS_KEY')
+            def client = containerClient(accountName, accountKey, containerName)
+            def bc = client.getBlobClient(topic)
+            def start = Instant.now()
+
+            def cnt = connectorContainer('azure_storage_blob_changefeed_source_0.1.json', [
+                'kafka_topic' : topic,
+                'kafka_bootstrap_servers': kafka.outsideBootstrapServers,
+                'kafka_consumer_group': group,
+                'azure_period': '2000',
+                'azure_account_name': accountName,
+                'azure_access_key': accountKey
+            ])
+
+            cnt.start()
+
+        when:
+            bc.upload(BinaryData.fromString(payload))
+
+        then:
+            kafka.poll(group, topic).any {
+                log.info("RECORD = {}", it.value())
+                return it.value() != null
+                        && !it.value().isEmpty()
```

Comment:
won't we validate the value?

Reply:
The value has nothing to do with what we send to the blobl it is something of this form:
```json
{  "api": "DeleteBlob",
    "clientRequestId": "9c2fbc6a-2f7b-4d64-84b5-d593e1a63881",
    "requestId": "1049d8d6-901e-0011-298f-100dd2000000",
    "contentType": "application/octetstream",
    "contentLength":36,
    "blobType":"BlockBlob",
    "contentOffset": null,
    "destinationUrl":null,
    "sourceUrl":null,
    "blobUrl":"https://tnbblob.blob.core.windows.net/engblob/f741f282-51b1-45dd-9826ffcb2a1900ba",
    "recursive":false,
    "sequencer":"000000000000000000000000000085A700000000018617ad",
    "etag":"0x8DADEA719BC0A02"
}
```
I am not sure how to validate it :confused: . Any ideas is welcome.

Reply:
I have added a sort of value validation: that is a json, that has a `blobUrl` property.

Reply:
This event seem to be a DeleteBlob event, but what you seem to be doing is writing to the Blob. Are we validating the correct event?

Reply:
I picked a random one from some logs I went through while developing it. It was just an example to show that almost nothing in the event is correlated with the content we send.